File: Entities/Models/Habit.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models;

public class Habit
{
    [Column("HabitId")]
    public Guid Id { get; set; }

    public bool IsArchived { get; set; }

    public int Color { get; set; }

    public string? Description { get; set; }

    public int FrequencyDensity { get; set; }

    public int FrequencyNumber { get; set; }

    public int Highlight { get; set; }

    [Required(ErrorMessage = "Name is a required field.")]
    public string? Name { get; set; }

    public int Position { get; set; }

    public DateTime ReminderTime { get; set; }

    public int ReminderDays { get; set; }

    public bool IsMeasurable { get; set; }

    public int Type { get; set; }

    public int TargetType { get; set; }

    public double TargetValue { get; set; }

    public string? Unit { get; set; }

    public string? Question { get; set; }
}

File: frontend/LoopHabitsNG/src/app/habits/habit.ts
import { Repetition } from "./repetition";

export interface Habit {
  id: string;
  isArchived: boolean;
  color: number;
  description: string;
  frequencyDensity: number;
  frequencyNumber: number;
  highlight: number;
  name: string;
  position: number;
  reminderTime: Date;
  reminderDays: number;
  isMeasurable: boolean;
  type: number;
  targetType: number;
  targetValue: number;
  unit: string;
  question: string;
  colorString: string;
  repetitions: Repetition[];
}
